src/lib/tls/tls_extensions.cpp

```diff
@@ -365,7 +365,8 @@ const std::vector<Group_Params>& Supported_Groups::groups() const {
 std::vector<Group_Params> Supported_Groups::ec_groups() const {
    std::vector<Group_Params> ec;
    for(auto g : m_groups) {
-      if(group_param_is_dh(g) == false) {
+      // This conditional assumes that we do not support a PQC hybrid with DH
+      if(g.is_diffie_hellman() == false && g.is_pure_kyber() == false) {
          ec.push_back(g);
       }
```

Comment:
Frankly, I'm not exactly sure what this does. I'm assuming "any group that uses ECC, either hybrid or not".
Should we promote this into a method on the new `Group_Params` class? E.g. We're planning to allow FrodoKEM for hybrid key exchange soon. I'm sure I'll forget to add `.is_pure_frodo()` here.

Reply:
I had made this change on the basis of "any group that uses ECC, hybrid or not" but on further checking, we only use this for TLS 1.2 negotiation where we should not attempt to negotiate a hybrid. So probably this should only return the list of standard ECDH curves, if any.

Reply:
Changed this to just filter using `is_ecdh_named_curve`